src/robotFactory.js

```diff
@@ -1,10 +1,76 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    coords.x = coords.x || 0;
+    coords.y = coords.y || 0;
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
+    this.coords = { ...coords };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    const { name, chipVersion, weight } = this;
+
+    return `Robot: ${name}, Chip version: ${chipVersion}, Weight: ${weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    coords.z = coords.z || 0;
+    super(name, weight, coords, chipVersion);
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords = {
+    x: 0,
+    y: 0,
+    z: 0,
+  },
+  chipVersion, maxLoadWeight, currentLoad = null) {
```

Comment:
You already have a default value for coordinates in the parent's classes

```suggestion
  constructor(
    name, 
    weight, 
    chipVersion, 
    maxLoadWeight, 
    currentLoad = null
  ) {
```